File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/resolvedmodel/expressionparser/PredefinedType.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.resolvedmodel.expressionparser;

/**
 * @deprecated This class is extremely likely to be removed in the public interface, and not
 * meant to be called externally at all. Please refrain from using it.
 */
@Deprecated
public enum PredefinedType {
    NotOperator,
    Operator,
    Token,
    Constant,
    OpenParenthesis,
    CloseParenthesis,
    Custom
}

File: objectModel/Python/cdm/resolvedmodel/expression_parser/predefined_type.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from enum import IntEnum


class PredefinedType(IntEnum):
    NOT_OPERATOR = 1
    OPERATOR = 2
    TOKEN = 3
    CONSTANT = 4
    OPEN_PARENTHESIS = 5
    CLOSE_PARENTHESIS = 6
    CUSTOM = 7
